### PR TITLE
fix: wrong escape with link, image url when converting

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description
* related issue
  * #1964 
  * #2099

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/editor/src/utils/common.ts [hunk 1]
```diff
@@ -12,6 +12,31 @@ const reEscapeHTML = /<([a-zA-Z_][a-zA-Z0-9\-._]*)(\s|[^\\/>])*\/?>|<(\/)([a-zA-
 const reEscapeBackSlash = /\\[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~\\]/g;
 const reEscapePairedChars = /[*_~`]/g;
 
+const XMLSPECIAL = '[&<>"]';
+const reXmlSpecial = new RegExp(XMLSPECIAL, 'g');
```
Comment (adhrinae) [inline apps/editor/src/utils/common.ts:16]: 이 정규표현식은 리터럴로 표현이 안되어서 `RegExp` 객체로 만드신건가요?

**Review by adhrinae — APPROVED** [review]
리뷰 완료합니다.

File: apps/editor/src/utils/common.ts (hunk 1 above)
Comment (js87zz) [reply apps/editor/src/utils/common.ts:16]: 저 문자열들을 따로 사용해서 정규식을 만드는 경우가 있을 수 있어서 이전에 마크다운 파서에서처럼 똑같이 분리해서 작성해두었습니다. 추후 따로 재사용성이 전혀 없다고 판단되면 그때 통합할게요~

**Comment by ibedarev** [comment]
Hi. Why do you need to escape `&`?

**Comment by js87zz** [comment]
@ibedarev 
It's mistake. It will be fixed.

**Comment by cary-hu** [comment]
Why not using encodeURI for url encode

**Pull request merged**